javaslang-benchmark/src/test/java/javaslang/benchmark/collection/TreeMapBenchmark.java

```diff
@@ -0,0 +1,398 @@
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object ceiling_Java(BenchmarkState state) {
+        return state.javaTreeMap.ceilingEntry(state.nextRandomKey());
+    }
+    @Benchmark
+    @Group("Navigate")
+    public Object higher_Slang(BenchmarkState state) {
+        return state.slangTreeMap.higher(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object higher_Java(BenchmarkState state) {
+        return state.javaTreeMap.higherEntry(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object lower_Slang(BenchmarkState state) {
+        return state.slangTreeMap.lower(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("Navigate")
+    public Object lower_Java(BenchmarkState state) {
+        return state.javaTreeMap.lowerEntry(state.nextRandomKey());
+    }
+
+    @Benchmark
+    @Group("IterateAscending")
+    public Object iterateAscending_Slang(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        for (Tuple2<Integer, String> entry : state.slangTreeMap) {
+            values.push(entry._1);
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateAscending")
+    public Object iterateAscending_Java(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        for (Map.Entry<Integer, String> aTreeMap : state.javaTreeMap.entrySet()) {
+            values.push(aTreeMap.getKey());
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateAscending")
+    public Object iterateAscending_Scala(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        Iterator<scala.Tuple2<Integer, String>> iterator = state.scalaTreeMap.iterator();
+        while (iterator.hasNext()) {
+            values.push(iterator.next()._1);
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateDescending")
+    public Object iterateDescending_Slang(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        for (Tuple2<Integer, String> aTreeMap : state.slangTreeMap.descendingIterator()) {
+            values.push(aTreeMap._1);
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("IterateDescending")
+    public Object iterateDescending_Java(BenchmarkState state) {
+        Stack<Integer> values = List.empty();
+        for (Map.Entry<Integer, String> aTreeMap : state.javaTreeMap.descendingMap().entrySet()) {
+            values.push(aTreeMap.getKey());
+        }
+        return values;
+    }
+
+    @Benchmark
+    @Group("SubMap")
+    public Object createSubMap_WithOneThirdElements_AndIterate_Slang(BenchmarkState state) {
+        // We will create subset with 1/3 of entries in the original map
+        int keyOffset = state.numItems / 6;
+        SortedMap<Integer, String> subMap = state.slangTreeMap.subMap(state.keysSorted.get(keyOffset), true, state.keysSorted.get(state.numItems - keyOffset), true);
```

Comment:
the primitive parameters seem confusing (`true`, ..., `true), could we extract them to constants with names representing their intentions?